### ISSUE TITLE
[BUG] Unable to save the value of a state variable from custom tool

### ISSUE THREAD

**Issue by vishwajeetio** [FlowiseAI/Flowise#3118]
**Describe the bug**
I am trying to save the value of a state variable on conditional agent from the custom js tool. but nothing seems to be working

**To Reproduce**

1. Go to a conditional agent
2. Click on custom js tool
3. Scroll down to the js code
4. See error by trying to save just a random value for a random variable to the state

**Expected behavior**
Should update the value of state variable

**Comment by HenryHengZJ** [comment]
can you share the flow? you can only save the state via Update State:
![image](https://github.com/user-attachments/assets/80016ea1-e746-461b-8353-c0894edd822b)

**Comment by takefy-dev** [comment]
how you can save state with a custom tool ?

**Comment by vishwajeetio** [comment]
I couldn't so decided to save the data from custom tool to a json and later on used it when ever needed. Hope that answers your question.

**Comment by takefy-dev** [comment]
yeah pretty work around, would be nice in the future to be able to update state everywhere in the flow

**Comment by vishwajeetio** [comment]
yeah, tried everything but that's the best I could find. You can assign the json name to chat id and then you can use it inside other tool with the chat id.

**Comment by veris-christopher** [comment]
@vishwajeetio could you provide detailed steps as to how you achieved this please? i am stuck on this for days as well.

**Comment by walldemar** [comment]
I got a similar issue:

![image](https://github.com/user-attachments/assets/6d0f75f9-0538-49f7-8e7c-71447266482f)

My workaround was:

![image](https://github.com/user-attachments/assets/148d9110-f12f-4bdc-9af6-40fc0dddf6bc)

**Comment by yuralves** [comment]
facing the same issue here. @vishwajeetio can you share details on your solution?

**Comment by thiagolealassis** [comment]
While @vishwajeetio  approach of saving data generated by the custom tool into a JSON file is a practical workaround, it introduces additional complexity and potential challenges in file management and data synchronization. A more efficient solution would be to enable custom tools to directly access and modify the $flow.state, allowing the setting of any necessary attributes from within the custom tool. This functionality would provide a more seamless integration, streamline development within the platform, and eliminate the need for alternative solutions, thereby enhancing the overall efficiency of the workflow.

@HenryHengZJ is this possible?

**Comment by HenryHengZJ** [comment]
With agentflow V2, you can now use Tool Node, LLM Node or Agent Node to update the flow state:

![Image](https://github.com/user-attachments/assets/cf0e3abc-1814-40de-b5e2-3fade95c43cf)
